### DISCUSSION TITLE
[LAMMPS] DPLR potential can't give the outputs after the initial step was conducted. HELP!!!

### DISCUSSION THREAD

**Discussion by gzp1314** [deepmodeling/deepmd-kit#2044] [Q&A] [answered]
the DPLR potential stops giving outputs after the initial step was done without any error meassages, altough i continue this jod for 5 hours. the training preformances of deep_wannier and DPLR model always give me confidence but the MD run was stuck after the initial step without any error messages giving off.  this situation drives me crazy.  the follwing is the tail of the output in lammps's logfile. 
''
  master list distance cutoff = 9.5
  ghost atom cutoff = 9.5
  binsize = 4.75, bins = 10 10 5
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin
      stencil: full/bin/3d
      bin: standard
Setting up Verlet run ...
  Unit style    : metal
  Current step  : 0
  Time step     : 0.0005
Per MPI rank memory allocation (min/avg/max) = 11.35 | 11.35 | 11.35 Mbytes
   Step         Volume          Temp          PotEng         KinEng         TotEng         Press
WARNING: Too many warnings: 291 vs 100. All future warnings will be suppressed (src/thermo.cpp:465)
         0   39851.779      0             -1160.9988      0             -1160.9988     -8006.5184
''

PS : the lammps i used is the built-in version in the deepmd-kit (version = 2.1.5). 
PS : I test my environment by sucessfully runing the example of water given by the hands-on sessions on DPLR (this is the link of it : [https://github.com/CSIprinceton/workshop-july-2022/tree/main/hands-on-sessions])

**Comment by wanghan-iapcm** [comment]
Could you please provide your lammps input script?

**Reply by gzp1314** [reply]
Thanks for your reply. 
You can find the configuration file and lammps script in the attachment. I also put the training scripts about the deep_wannier and DPLR model in it. 

Update:
I just fail to get the result when i use the follwing python script to check the dipole distribution of my atomic configuration by my deep_wannier model. This information may be help. 
"
from deepmd.infer import DeepDipole 
import ase.io.vasp
import numpy as np

f = './3/4168/'
dipole_model = DeepDipole('./diople.pb')

configs = ase.io.vasp.read_vasp(f+"/POSCAR")

atypes = np.array(configs.get_atomic_numbers()) - 1

dipole = dipole_model.eval(configs.get_positions(),configs.get_cell(),atom_types=atypes)

"
[SCRIPTS.zip](https://github.com/deepmodeling/deepmd-kit/files/9915253/SCRIPTS.zip)

**Comment by wanghan-iapcm** [comment, marked as answer]
By your script you want to associate virtual atom type 3 to real atom type 2, but the .lmp file is linking atoms type3 to type 1.  

PS: We recommend setting the coordinates of the virtual atoms the same as the real atoms to which it associated.

**Reply by gzp1314** [reply]
I must say sorry for my mistake to provide a wrong version of data.lmp. I already corrcted this few days ago. However, the same question still.  This is the current data.lmp. Unfortunately, with this data file and the same lammps script, the MD still refuse to give me output after the initial step.



![image](https://user-images.githubusercontent.com/61399060/199880727-b8871b07-5724-4ff6-94e8-400e5eb07f5d.png)

PS : The purple stick is represent for the bonds. Now the virtual atom (type 3) correctly associate to the real atom (type 2 )

"
# LAMMPS data file written by OVITO Basic 3.7.4
54 atoms
3 atom types
18 bonds
1 bond types
0.0 14.323764456 xlo xhi
0.0 15.456744154 ylo yhi
0.0 20.0 zlo zhi
0.0 0.015286353 0.006080383 xy xz yz

Masses

1 207.2  # Pb
2 127.6  # Te
3 127.6  # X

Atoms  # full

1	1	1	2	13.94989	2.15493	8.51626
2	1	1	2	0.24207	7.26525	8.58236
3	1	1	2	0.15916	13.4256	9.20057
4	1	1	2	5.54125	2.98099	8.54042
5	1	1	2	4.26232	7.2625	8.37701
6	1	1	2	4.87457	12.91533	8.7699
7	1	1	2	9.58226	2.56973	8.4505
8	1	1	2	9.74629	7.89716	8.78916
9	1	1	2	9.42717	11.88904	8.47521
10	1	1	2	3.02151	0.46799	11.32202
11	1	1	2	2.09971	4.462	11.43385
12	1	1	2	2.56724	10.01677	11.69191
13	1	1	2	7.60635	15.3216	11.26812
14	1	1	2	7.35365	5.54337	10.98555
15	1	1	2	6.63247	9.60885	11.70394
16	1	1	2	11.56079	14.80505	11.57819
17	1	1	2	12.16843	5.12687	11.53209
18	1	1	2	12.30139	10.35047	11.40212
19	1	2	6	14.03871	2.47346	11.55175
20	1	2	6	0.11287	8.11184	11.57619
21	1	2	6	13.95946	13.00409	12.16964
22	1	2	6	5.00136	3.22749	11.55536
23	1	2	6	4.36581	7.41253	11.39038
24	1	2	6	5.00799	13.50131	11.71275
25	1	2	6	9.55782	3.23548	11.3682
26	1	2	6	9.20674	8.0407	11.76647
27	1	2	6	9.21337	12.705	11.35689
28	1	2	6	2.31962	0.49252	8.37993
29	1	2	6	2.24438	4.95096	8.41343
30	1	2	6	2.33173	11.01319	8.82926
31	1	2	6	7.28501	0.43129	8.23732
32	1	2	6	6.97837	5.80314	7.94196
33	1	2	6	6.80557	10.31466	8.74035
34	1	2	6	11.78998	15.36115	8.51581
35	1	2	6	11.94364	5.56808	8.55297
36	1	2	6	12.0343	10.37308	8.41946
37	1	3	-8	14.08434134	2.64653618	11.36588609
38	1	3	-8	0.128973876	8.213134505	11.40918083
39	1	3	-8	13.81206715	13.01521547	11.905815
40	1	3	-8	4.886063683	3.19141205	11.34439811
41	1	3	-8	4.407133588	7.57302667	11.23523174
42	1	3	-8	5.013297788	13.65286493	11.48333655
43	1	3	-8	9.565371448	3.338126793	11.15672568
44	1	3	-8	9.067783288	8.020554828	11.54011384
45	1	3	-8	9.2735376	12.86216585	11.19784527
46	1	3	-8	2.206231955	0.468200139	8.6252808
47	1	3	-8	2.233673708	5.125573938	8.59355138
48	1	3	-8	2.390679628	11.10217932	9.024382908
49	1	3	-8	7.319796113	0.61674052	8.449932988
50	1	3	-8	6.875238583	5.742005995	8.183818348
51	1	3	-8	6.86667362	10.4088793	8.90348149
52	1	3	-8	11.86036383	15.47757568	8.703367773
53	1	3	-8	11.9924759	5.682586195	8.75019083
54	1	3	-8	11.85851064	10.36812398	8.63789749



Bonds

1	1	19	37
2	1	20	38
3	1	21	39
4	1	22	40
5	1	23	41
6	1	24	42
7	1	25	43
8	1	26	44
9	1	27	45
10	1	28	46
11	1	29	47
12	1	30	48
13	1	31	49
14	1	32	50
15	1	33	51
16	1	34	52
17	1	35	53
18	1	36	54

"

**Reply by wanghan-iapcm** [reply]
Please try setting the coordinates of the virtual atoms the same as the real atoms to which it associated.

**Reply by gzp1314** [reply]
Thank you very much !!!. I'm so happy it is fixd after i set the same coordinates about the real atoms and the corrsponding virtual atoms.  
But i'm still wondering why it must be the same coordinates between the real atoms and the corrsponding virtual atoms. This constraint feels a little SPECIAL

**Reply by wanghan-iapcm** [reply]
The virutal atoms should be located to the same subdomain as its associated real atom during the domain decomposition. This can only be safely guaranteed if they have the same coordinate.

**Reply by gzp1314** [reply]
Thanks, it helps me a lot.
